sync/class.jetpack-sync-actions.php

```diff
@@ -178,18 +182,58 @@ static function send_data( $data, $codec_name, $sent_timestamp, $queue_id ) {
 	}
 
 	static function do_initial_sync( $new_version = null, $old_version = null ) {
-		$initial_sync_config = array(
+		$initial_sync_config = self::get_update_full_sync_config();
+		$include_users = false;
+		if ( $old_version && ( version_compare( $old_version, '4.2', '<' ) ) ) {
+			$initial_sync_config['users'] = 'initial';
+			$include_users = true;
+		}
+
+		if ( is_multisite()  ) {
+			// stagger initial syncs for multisite blogs so they don't all pile on top of each other
+			if ( is_main_site() ) {
+				wp_schedule_single_event( time() , 'jetpack_sync_update_multisite_cron', $include_users );
+			}
+		} else {
+			self::do_full_sync( $initial_sync_config );
+		}
+	}
+
+	static function get_update_full_sync_config() {
+		return array(
 			'options' => true,
 			'network_options' => true,
 			'functions' => true,
 			'constants' => true,
 		);
-
-		if ( $old_version && ( version_compare( $old_version, '4.2', '<' ) ) ) {
+	}
+	
+	static function update_multisite_full_sync( $include_users = false ) {
+		global $wpdb;
+		$initial_sync_config = self::get_update_full_sync_config();
+		if ( $include_users ) {
 			$initial_sync_config['users'] = 'initial';
 		}
 
-		self::do_full_sync( $initial_sync_config );
+		$offset = 0;
+		$batch = 500;
+		$continue = true;
+
+		while( $continue ) {
```

Comment:
why not just say `while ( true )` here? You never change the value of `$continue`